### DISCUSSION TITLE
How to stream binary / byte response?

### DISCUSSION THREAD

**Discussion by lanyusan** [http4s/http4s#6477] [Q&A]
We are trying to migrate a clojure based web server (pedestal) to http4s. 
We have an endpoint that streams back some data as `"application/octet-stream`
We used `InputStream` with the Pedestal legacy code.
While with http4s I use the following code:

```
val fs2InStream = readInputStream(IO(inStream), bufferSize)
 // Ok(fs3inStream)   -- was a typo
Ok(fs2InStream
```
The `inStream` is Java `InputStream`.

The compiler tells:
```
Cannot convert from Object to an Entity, because no EntityEncoder[cats.effect.IO, Object] instance could be found.
```

I have imported decoders:
```
import org.http4s.EntityEncoder._
```

Can I get some advise here?
Thanks.

**Comment by armanbilge** [comment]
Can you post a more complete code snippet? The one you posted has some typos `fs2InStream` vs `fs3inStream` so it's difficult to identify the problem :)

**Reply by lanyusan** [reply]
Sorry It is a typo in the question. 

```
val fs2InStream = readInputStream(IO(inStream), bufferSize)
Ok(fs2InStream, `Content-Type`(MediaType.application.`octet-stream`))

```

**Comment by lanyusan** [comment]
@armanbilge 
hello, any feedback on this? Or an example of streaming `octet-stream` response would help as well.

**Reply by armanbilge** [reply]
This compiles fine for me:

https://scastie.scala-lang.org/vx9yNMFsRTqowggtyNfxVA
```scala
import cats.effect._
import org.http4s.Status._
import org.http4s.dsl.io._
import fs2.io.readInputStream

def inStream: java.io.InputStream = ???
val bufferSize: Int = 8192
val fs2InStream = readInputStream(IO(inStream), bufferSize)
Ok(fs2InStream)
```

**Reply by lanyusan** [reply]
Thanks for the response. Not sure what's wrong with our code. I will further look into it.
